unity/Assets/Maroon/scenes/experiments/PlanetarySystem/Scripts/SortingPlanetsDragAndDrop.cs

```diff
@@ -0,0 +1,161 @@
+        private readonly float scaleFactor = 1.02f;
+        public Light sunLightHalo;
+        Vector3 mousePosition;
+
+        public TextMeshProUGUI planetInfoMessageText;
+        string createdPlanetInfoMessage;
+
+
+        //handle mouse input
+        #region MouseInput
+        /// <summary>
+        /// get mouse position
+        /// </summary>
+        private void OnMouseDown()
+        {
+            audioSource.PlayOneShot(pickUpClip);
+            mousePosition = Input.mousePosition - Camera.main.WorldToScreenPoint(transform.position);
+            // Show PlanetInfo data when planet is clicked on
+
+            PlanetInfo planetInfo = GetComponent<PlanetInfo>();
+            createdPlanetInfoMessage = planetInfo.CreateUnnamedPlanetInfoMessage();
+            planetInfoMessageText.text = createdPlanetInfoMessage;
+        }
+
+
+        /// <summary>
+        /// calculate position
+        /// </summary>
+        private void OnMouseDrag()
+        {
+            if (!isSnapped)
+            {
+                Vector3 newPosition = Camera.main.ScreenToWorldPoint(Input.mousePosition - mousePosition);
+                transform.position = newPosition;
+            }
+        }
+
+
+        /// <summary>
+        /// snap to target when distance is smaller han snap distance
+        /// or snap back to start slot
+        /// </summary>
+        private void OnMouseUp()
+        {
+            if (!isSnapped)
+            {
+                Vector2 distanceVector = new Vector2(transform.position.x - sortingPlanetTarget.position.x, transform.position.y - sortingPlanetTarget.position.y);
+
+                //Debug.Log("SortingPlanetsDragAndDrop(): distanceVector.magnitude: " + distanceVector.magnitude);
+                if (distanceVector.magnitude <= snapDistance)
+                {
+                    SnapToTarget();
+                }
+                else
+                {
+                    transform.position = transform.parent.position;
+                    isSnapped = false;
+                    PlanetInfo planetInfo = GetComponent<PlanetInfo>();
+                    planetInfo.IsSnapped = false;
+                }
+            }
+        }
+        #endregion MouseInput
+
+
+        //snap the planets to target
+        #region SnapPlanet
+        /// <summary>
+        /// increments the snapped planets and displays a Helpi message when all planets except pluto are in its place 
+        /// </summary>
+        private void IncrementSnappedPlanetCount()
+        {
+            int solarSystemPlanet = 10;
+            planetSortingGameController.sortedPlanetCount++;
+
+            // check if we have snapped all 10 out of 11 planets, excluding pluto, including moon ad sun
+            if (planetSortingGameController.sortedPlanetCount >= solarSystemPlanet)
+            {
+                PlanetaryController.Instance.DisplayMessageByKey("OrderedSortingGame");
+            }
+        }
+
```

Comment:
I think this method would be more suited to be in the planetary controller class